src/ts/Offers/index.tsx

```diff
@@ -96,12 +101,36 @@ export default class Offers extends React.Component<any, any> {
     this.state = {
       data: [],
       loading: true,
+      networkErr: false,
+      verifyError: false,
+      verifySuccess: false,
     }
-    this.loadMore = this.loadMore.bind(this);
+    this.loadMore = this.loadMore.bind(this)
+    this.verifyAccount = this.verifyAccount.bind(this)
   }
 
   componentDidMount() {
-    this.loadMore();
+    this.loadMore()
+    if (this.props.params.token) {
+      this.verifyAccount()
+    }
+  }
+
+  verifyAccount() {
```

Comment:
this function does not belong in the offers component. It should probably be higher up in the hierarchy